CST-to-AST/convert-special.lisp

```diff
@@ -430,12 +436,15 @@
       ;; FIXME: We probably want to create and use
       ;; env:constantp in case the environments don't match
       ;; up.
+      ;; FIXME(paul) What is env:constantp?
```

Comment:
`env:constantp` would be a `cleavir-env` function that's like `cl:constantp`, only a customizable generic. This is out of scope for Trucler since it could be quite sophisticated in general, e.g. detecting that `(let () 4)` is a constant. But a minimal implementation would do `cl:constantp`'s minimum, i.e. it can detect self-evaluating objects, quoted forms, and constant variables (which would be found via Trucler). Given that you've deleted `cleavir-env`, this function would be in `cst-to-ast` instead.